pkg/pubsub/adapter/converters/pubsub.go

```diff
@@ -58,5 +62,74 @@ func convertPubSub(ctx context.Context, msg *cepubsub.Message, sendMode ModeType
 			}
 		}
 	}
+	// If send mode is Push, convert to Pub/Sub Push payload style.
+	if sendMode == Push {
```

Comment:
this if should have the else clause, where you would have the msg.Attributes promoted to extensions as in the above lines... only promote attributes to extensions if it's not push...

Reply:
done, note as I mentioned below, tests used to check the event had the extensions as well in Push mode.